kview/src/main/java/dev/ikm/komet/kview/controls/skin/KLReadOnlyComponentSetControlSkin.java

```diff
@@ -0,0 +1,181 @@
+            {
+                super.bind(control.titleProperty());
+            }
+            @Override
+            protected String computeValue() {
+                String title = control.getTitle();
+                if (title != null) {
+                    return control.getTitle().toUpperCase();
+                } else {
+                    return "";
+                }
+            }
+        });
+
+        // sync items observableSet
+        for (ComponentItem componentItem : control.getItems()) {
+            addNewUIItem(componentItem);
+        }
+        control.getItems().addListener(this::itemsChanged);
+
+        // CSS
+        mainContainer.getStyleClass().add("main-container");
+        componentsContainer.getStyleClass().add("components-container");
+        titleLabel.getStyleClass().add("title");
+        promptTextLabel.getStyleClass().add("prompt-text");
+    }
+
+    private void itemsChanged(SetChangeListener.Change<? extends ComponentItem> change) {
+        if (change.wasAdded()) {
+            addNewUIItem(change.getElementAdded());
+        } else if (change.wasRemoved()) {
+            removeUIItem(change.getElementRemoved());
+        }
+    }
+
+    private void addNewUIItem(ComponentItem componentItem) {
+        Node componentUIItem = new ComponentItemNode(this, componentItem);;
+        componentsContainer.getChildren().add(componentUIItem);
+        componentUIItems.put(componentItem, componentUIItem);
+    }
+
+    private void removeUIItem(ComponentItem componentItem) {
+        Node componentUIItem = componentUIItems.get(componentItem);
+        componentsContainer.getChildren().remove(componentUIItem);
+        componentUIItems.remove(componentItem);
+    }
+
+    private static MenuItem createMenuItem(String text, KometIcon.IconValue icon, EventHandler<ActionEvent> actionHandler) {
+        MenuItem menuItem = new MenuItem(text, KometIcon.create(icon, "icon-klcontext-menu"));
+        menuItem.setOnAction(actionHandler);
+        return menuItem;
+    }
+
+    protected void fireOnEditAction(ActionEvent actionEvent) {
+        if (getSkinnable().getOnEditAction() != null) {
+            getSkinnable().getOnEditAction().run();
+        }
+    }
+
+    protected void fireOnRmoveAction(ActionEvent actionEvent, ComponentItem componentItem) {
+        if (getSkinnable().getOnRemoveAction() != null) {
+            getSkinnable().getOnRemoveAction().accept(componentItem);
+        }
+    }
+
+    /*******************************************************************************
+     *                                                                             *
+     * Supporting Classes                                                          *
+     *                                                                             *
+     ******************************************************************************/
+
+    public static class ComponentItemNode extends Region {
```

Comment:
Could this class be pulled out so it can also be used in Component List?

Reply:
Or are they different. E.g. Set does not have (numbers/index) beside them like for Component List.